src/components/user-settings-form.jsx

```diff
@@ -19,9 +20,14 @@ export default class UserSettingsForm extends React.Component {
   render() {
     return (
       <form onSubmit={preventDefault(this.updateUser)}>
-        <div className="form-group">
+        <div className={classnames({'form-group': true, 'has-feedback': this.props.screenName, 'has-error': this.props.screenName && (this.props.screenName.length < 3 || this.props.screenName.length > 25), 'has-success': this.props.screenName && (this.props.screenName.length >= 3 && this.props.screenName.length <= 25) })}>
```

Comment:
It would be better to extract the class builder in a separate expression before 'return'. Such long line is very difficult to read. See https://github.com/FreeFeed/freefeed-react-client/blob/stable/src/components/post-comment.jsx#L93 for example.